lib/audit/auditor.go

```diff
@@ -232,37 +236,23 @@ func (s *ServerContext) Audit(w http.ResponseWriter, r *http.Request) {
 		http.Error(w, err.Error(), http.StatusInternalServerError)
 		return
 	}
-	// Find the subproject's registry.
-	var srcRegistry reg.RegistryContext
-	for _, rc := range sc.RegistryContexts {
-		if !rc.Src {
-			continue
-		}
-		rcNameParts := strings.Split(string(rc.Name), "/")
-		if len(rcNameParts) < 3 {
-			continue
-		}
-		// Find the subproject key from the pub/sub message.
-		klog.Infof("subproject name: %s", rcNameParts[2])
-		if strings.HasSuffix(string(rc.Name), rcNameParts[2]) {
-			rcBound := rc // Avoid loop reference variable.
-			srcRegistry = rcBound
-			break
-		}
-	}
-	// If we can't find the source registry for this image, then reject the
+	// Find the subproject repository responsible for the GCRPubSubPayload. This
```

Comment:
It might be nice to split out PubSub (a transport) from the idea of a "Change" we are validating.  You could also then split this function into two, one that parses pubsub and one that validates a change.